provider/common/bootstrap.go

```diff
@@ -492,7 +492,7 @@ func hostBootstrapSSHOptions(
 	}()
 
 	options := &ssh.Options{}
-	options.SetStrictHostKeyChecking(ssh.StrictHostChecksYes)
+	options.SetStrictHostKeyChecking(ssh.StrictHostChecksNo)
```

Comment:
This either needs to be a feature flag or a bootstrap flag. This is so easy to miss when reviewing a large patch and landing this would cause all sorts of chaos.

Reply:
Thanks for calling this out, @SimonRichardson. 

I believe this was used while debugging, leading up to the need for https://github.com/juju/juju/pull/12349/files#diff-991a5d09268caffa5957891ac19ed4e9e5935d558c46064d48825bff7b68eedbR234.

Essentially, the Equinix Metal cloud-init scripts (on-disk) preempt Juju's SSH provisioning step, https://github.com/tinkerbell/osie/blob/71da59e10bf66cc352366615060171939412d397/docker/scripts/osie.sh#L360-L379.

@jasmingacic and I are aware of this and it won't be included when the Draft and WIP flags are removed.

Reply:
That makes sense. But we should really take a look into how Equinix is injecting the keys.

Reply:
Can we try removing this change now that the custom userdata is being passed in?

Reply:
I tried switching back and it is still failing.